v2/failure-injected-spanner-service/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerServiceFactoryImpl.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (C) 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.beam.sdk.io.gcp.spanner;
+
+import com.google.cloud.teleport.v2.spanner.service.SpannerService;
+
+public class SpannerServiceFactoryImpl {
```

Comment:
We already use `AutoValue` in the code, is it also possible to use `AutoFactory`? There could be opportunity to avoid boilerplate factory code.

Ref - https://www.baeldung.com/autofactory

Reply:
This is not actually a factory, this is a placeholder class in apache beam package in order to access methods in spannerConfig. The policyFactory class can use AutoFactory, I'll do it in the next PR when there are more policy implementations.